add zod support for MachineEvent (#59)
Actyx/machines: machine-runner/README.md

Machine runner resolves this conflict by using only the `selected` event that comes first in the Actyx event sort order; in other words, Actyx arbitrarily picks a winner and the losing event is discarded.
If a robot saw itself winning, started the mission, and then discovers that its win turned out to be invalid, it will have to stop the mission and pick a new one.

## Features

### Zod on MachineEvent

[Zod](https://zod.dev/) can be used to define and validate MachineEvents. On designing an event, use `withZod` instead of `withPayload`.

```typescript
export const requested = MachineEvent.design('requested').withPayload<{
  id: string
  from: string
  to: string
}>()
```

The above code can be converted into:

```typescript
import * as z from 'zod'

export const requested = MachineEvent.design('requested').withZod(
  z.object({
    id: z.string(),
    from: z.string(),
    to: z.string(),
  }),
)
```

> A zod-enabled event factories will have these additional features enabled:

Reviewer: ```suggestion
A zod-enabled event factory will have these additional features enabled:
```

> - When receiving events from Actyx, a `MachineRunner` will compare the event payload to the embedded `ZodType`, in addition to the mandatory event type checking. Events that don't match the previously inputted `MachineEvent` on the reaction definition will be ignored by the `MachineRunner`. For example, see the reaction definition below:

Reviewer 2: ```suggestion
- When receiving events from Actyx, a `MachineRunner` will compare the event payload to the embedded `ZodType`, in addition to the mandatory event type checking. Events that don't match the defined `MachineEvent` on the reaction will be ignored by the `MachineRunner`. For example, see the reaction definition below:
```

Author: Fixed

  ```typescript
  InitialWarehouse.react([requested], DoneWarehouse, (_ctx, _r) => [{}])
  ```

>   In a scenario where an incorrectly created event comes from Actyx `{ "type": "requested", id: "some_id" }`, the said event will not be regarded as valid and will be ignored.

Reviewer: ```suggestion
  In a scenario where an incorrectly created event comes from Actyx `{ "type": "requested", id: "some_id" }`, said event will not be regarded as valid and will be ignored.
```

> - When creating event via the factory, which would be `requested.make` for the example above, an extra step will be taken to validate the payload. When the `make` method is called with an incorrect value, an exception will be thrown because internally `ZodType.parse` is used to validate the payload. For example:

Reviewer: ```suggestion
- When creating the event via the factory, which would be `requested.make` for the example above, an extra step will be taken to validate the payload. When the `make` method is called with an incorrect value, an exception will be thrown because internally `ZodType.parse` is used to validate the payload. For example:
```

  ```typescript
  // Will throw because `{}` is not a valid value for the previously provided zod schema
  // But it takes `as any` to bypass TypeScript compiler in order to do this
  const singleEvent = requested.make({} as any)
  ```

>   An extra care must be put when the `ZodType` is [refined](https://zod.dev/?id=refine). Compared to mismatch schema, a refined `ZodType` is not caught at compile-time. Therefore, a compilation process and IDE warnings is not sufficient to catch this errors. For example:

Reviewer: ```suggestion
  An extra care must be put when the `ZodType` is [refined](https://zod.dev/?id=refine). Compared to mismatch schema, a refined `ZodType` is not caught at compile-time. Therefore, a compilation process and IDE warnings is not sufficient to catch these errors. For example:
```

>   An extra care must be put when the `ZodType` is [refined](https://zod.dev/?id=refine). Compared to mismatch schema, a refined `ZodType` is not caught at compile-time. Therefore, a compilation process and IDE warnings is not sufficient to catch these errors. For example:

Reviewer 2: ```suggestion
  An extra care must be taken when the `ZodType` is [refined](https://zod.dev/?id=refine). In contrast to a mismatch in schema, a refined `ZodType` is not caught at compile-time. Therefore, a compilation process and IDE warnings is not sufficient to catch these errors. For example:
```

Author: Fixed
